tests/extmod/ujson_dumps.py

```diff
@@ -1,8 +1,10 @@
 try:
     import ujson as json
+    from ucollections import OrderedDict
```

Comment:
Some ports might have `ujson` enabled but not `collections.OrderedDict`, and in those cases, the existing test here would still be valid, up until the OrderedDict check. Would it make sense to pull the OrderedDict testing into a different test case?

Reply:
With this test being one that's run in comparison to cpython I never considered it might get run on ports other than Unix, but it makes sense it could very easily (via pyboard.py etc).

Reply:
Even just looking at the `run-tests` script, tests under `extmod` will be run on every officially-supported target. https://github.com/micropython/micropython/blob/master/tests/run-tests#L588

The only officially-maintained port or build that ends up with ujson enabled but `collections.OrderedDict` disabled looks to be a debug build of cc3200. However, the reason I call out "some ports" is that I help maintain a fork/port with precisely that configuration, and we run the upstream tests against our devices to check compatibility. We _could_ remove this skip in our fork, but then that means we have to review each and every test change that lands to make sure no gaps arise.

Reply:
Moving to a separate file does make sense to me, let the standard "SKIP" functionality of the test suite handle cases where it's not available.

I've done this in my latest push